src/Microsoft.AspNet.Mvc.ModelBinding/Binders/FormFileModelBinder.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Mvc.ModelBinding.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    public class FormFileModelBinder : IModelBinder
+    {
+        public async Task<bool> BindModelAsync([NotNull] ModelBindingContext bindingContext)
+        {
+            if (bindingContext.ModelType == typeof(IFormFile))
```

Comment:
No, we want to bind to specific types here. In the other case you are likely to have `List<IFormFile>` or `IFormFile[]` etc which is why the `IsAssignableFrom` check is required.